File: Data Structures & Algorithms/max-water-container/submission-0.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int leftIndex = 0;
        int rightIndex = heights.Length - 1;
        int maxWater = 0;

        while(leftIndex < rightIndex){
            int area = (rightIndex - leftIndex) * Math.Min(heights[leftIndex], heights[rightIndex]);
            maxWater = Math.Max(maxWater, area);
            if(heights[leftIndex] < heights[rightIndex]){
                leftIndex++;
            }else{
                rightIndex--;
            }
        }

        return maxWater;
    }
}

File: Data Structures & Algorithms/max-area-of-island/submission-0.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {
        int maxArea = 0;

        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[i].length; j++){
                if(grid[i][j] == 1){
                    maxArea = Math.max(maxArea, maxAreaBFS(grid, i, j));
                }
            }
        }
        return maxArea;
    }

    public int maxAreaBFS(int[][] grid, int i, int j){
        if(i < 0 || i >= grid.length || j < 0 || j >= grid[i].length || grid[i][j] == 0){
            return 0;
        }

        grid[i][j] = 0;

        return maxAreaBFS(grid, i + 1, j) + maxAreaBFS(grid, i, j + 1) + maxAreaBFS(grid, i - 1, j) + maxAreaBFS(grid, i, j - 1) + 1;

    }
}
